src/core/thread/router_table.hpp

```diff
@@ -29,6 +29,8 @@
 #ifndef ROUTER_TABLE_HPP_
 #define ROUTER_TABLE_HPP_
 
+#if OPENTHREAD_FTD
```

Comment:
Not related to this PR, but reading this notcied that this file seems to be missing `#include "openthread-core-config.h"` at the top.
I think we should include it before the `#if OPENTHREAD_FTD` check?